src/common/bitstream.cpp

```diff
@@ -0,0 +1,392 @@
+#include <algorithm>
+#include "bitstream.h"
+#include "util.h"
+
+#define BITS_TO_BYTES(bits) (((bits) + 7) / 8)
+
+
+BitStream::BitStream(std::vector<uint8_t>& exisitingBuf) :
+    buf(exisitingBuf),
+    streamBitPos(0),
+    lastError(Error::NONE) {
+
+}
+
+size_t BitStream::getSizeBits() const {
+    return buf.size() * 8;
+}
+
+size_t BitStream::getRemainingBits() const {
+    size_t sizeBits = getSizeBits();
+    if(streamBitPos >= sizeBits) {
```

Comment:
We should keep our spaces after control flow statements consistent.
Tools->Options->Text Editor->C/C++->Formatting->Spacing->Spacing for control blocks->check Insert space between keyword... in VC++ will keep the space. I believe this is the default option anyway, unless you are using a different version of VC++ or something.